client/src/app-components/ticket-viewer.js

```diff
@@ -510,23 +522,47 @@ class TicketViewer extends React.Component {
     }
 
     addTag(tag) {
+        this.setState({
+            tagSelectorLoading: true,
+        })
         API.call({
             path: '/ticket/add-tag',
             data: {
                 ticketNumber: this.props.ticket.ticketNumber,
                 tagId: tag
             }
-        }).then(this.onTicketModification.bind(this))
+        })
+        .then(() => {
```

Comment:
.then should be next to the closing of the API.call in the same line.Put together line 534 and 535